src/realm/util/file.cpp

```diff
@@ -680,20 +680,50 @@ void File::resize(SizeType size)
 }
 
 
-void File::prealloc(SizeType offset, size_t size)
+void File::prealloc(size_t size)
 {
     REALM_ASSERT_RELEASE(is_attached());
 
 #if defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200112L // POSIX.1-2001 version
 
-    prealloc_if_supported(offset, size);
+    prealloc_if_supported(0, size);
 
 #else // Non-atomic fallback
 
-    if (int_add_with_overflow_detect(offset, size))
-        throw std::runtime_error("File size overflow");
-    if (get_size() < offset)
-        resize(offset);
+#if REALM_IOS
+    fstore_t store = { F_ALLOCATEALL, F_PEOFPOSMODE, 0, static_cast<off_t>(size), 0 };
+    int ret = fcntl(m_fd, F_PREALLOCATE, &store);
+    if (ret == -1) {
+        int err = errno;
+        std::string msg = get_errno_msg("fcntl() inside prealloc() failed: ", err);
+    }
+    ret = ftruncate(m_fd, size);
```

Comment:
Looks like `ftruncate` on apple can fail with `EINTR`. In this case I think we should try again in a loop instead of failing hard.